container/pull.bzl

```diff
@@ -84,20 +99,15 @@ container_import(
     if result.return_code:
         fail("Pull command failed: %s (%s)" % (result.stderr, " ".join(args)))
 
-container_pull = repository_rule(
-    attrs = {
-        "registry": attr.string(mandatory = True),
-        "repository": attr.string(mandatory = True),
-        "digest": attr.string(),
-        "tag": attr.string(default = "latest"),
-        "_puller": attr.label(
-            executable = True,
-            default = Label("@puller//file:downloaded"),
-            cfg = "host",
-        ),
-    },
-    implementation = _impl,
-)
+    updated_attrs = {
+        k: getattr(repository_ctx.attr, k)
+        for k in _container_pull_attrs.keys()
+    }
+    digest_result = repository_ctx.execute(["cat", repository_ctx.path("image/digest")])
+    if digest_result.return_code:
+        fail("Failed to read digest: %s" % digest_result.stderr)
+    updated_attrs["digest"] = digest_result.stdout
+    return updated_attrs
```

Comment:
How is this used? "bazel sync" will pick this up?

Reply:
Yes, that is my understanding. Future versions of bazel add the `bazel sync` command as well as experimental flags which save the results in a specified `.bzl` file, including the return values of any repository rules; other flags would let you tell bazel to read workspace dependencies from this resolved `.bzl` file instead of from `WORKSPACE`. [This blog post](https://blog.bazel.build/2018/07/09/bazel-sync-and-resolved-file.html) gives a better explanation.

Reply:
Cool. Thanks for the explanation and pointer.